File: src/examples/InternationalStreetExample.cs
namespace Examples
{
	using System;
    using System.IO;
    using System.Net;
    using System.Threading.Tasks;
    using SmartyStreets;
	using SmartyStreets.InternationalStreetApi;

	internal static class InternationalStreetExample
	{
		public static async Task Run()
		{
            // specifies the TLS protocoll to use - this is TLS 1.2
            const SecurityProtocolType tlsProtocol1_2 = (SecurityProtocolType)3072;

            // We recommend storing your secret keys in environment variables.
            var authId = Environment.GetEnvironmentVariable("SMARTY_AUTH_ID");
			var authToken = Environment.GetEnvironmentVariable("SMARTY_AUTH_TOKEN");
			ServicePointManager.SecurityProtocol = tlsProtocol1_2;

			var client = new ClientBuilder(authId, authToken).BuildInternationalStreetApiClient();
			
			// Documentation for input fields can be found at:
			// https://smartystreetscom/docs/cloud/international-street-api#http-input-fields

			// Geocoding must be expressly set to get latitude and longitude.
			var lookup = new Lookup("Rua Padre Antonio D'Angelo 121 Casa Verde, Sao Paulo", "Brazil")
			{
				InputId = "ID-8675309", // Optional ID from your system
				Geocode = true,
				Organization = "John Doe",
				Address1 = "Rua Padre Antonio D'Angelo 121",
				Address2 = "Casa Verde",
				Locality = "Sao Paulo",
				AdministrativeArea = "SP",
				Country = "Brazil",
				PostalCode = "02516-050"
			};
			
			//uncomment the line below to add a custom parameter
			//lookup.AddCustomParameter("input_id", "ID-8675309");

            try
            {
                await client.Send(lookup);
            }
            catch (SmartyException ex)
            {
                Console.WriteLine(ex.Message);
                Console.WriteLine(ex.StackTrace);
                return;
            }
            catch (IOException ex)
            {
                Console.WriteLine(ex.StackTrace);
                return;
            }

            var candidates = lookup.Result;

            if (candidates.Count == 0)
            {
                Console.WriteLine("No candidates. This means the address is not valid.");
                return;
            }

            var firstCandidate = candidates[0];
			Console.WriteLine("Input ID: " + firstCandidate.InputId);
			Console.WriteLine("Address is " + firstCandidate.Analysis.VerificationStatus);
			Console.WriteLine("Address precision: " + firstCandidate.Analysis.AddressPrecision + "\n");

			Console.WriteLine("First Line: " + firstCandidate.Address1);
			Console.WriteLine("Second Line: " + firstCandidate.Address2);
			Console.WriteLine("Third Line: " + firstCandidate.Address3);
			Console.WriteLine("Fourth Line: " + firstCandidate.Address4);
			Console.WriteLine("Address Format: " + firstCandidate.Metadata.AddressFormat);
			Console.WriteLine("Latitude: " + firstCandidate.Metadata.Latitude);
			Console.WriteLine("Longitude: " + firstCandidate.Metadata.Longitude);
		}
	}
}

File: src/vbexamples/InternationalStreetExample.vb
Imports System
Imports System.Diagnostics.Metrics
Imports System.IO
Imports System.Windows
Imports SmartyStreets
Imports SmartyStreets.InternationalStreetApi

Module InternationalStreetExample

	Dim authID = Environment.GetEnvironmentVariable("SMARTY_AUTH_ID")
	Dim authToken = Environment.GetEnvironmentVariable("SMARTY_AUTH_TOKEN")

	Dim client = New ClientBuilder(authID, authToken).WithLicense(New List(Of String) From {"international-global-plus-cloud"}).BuildInternationalStreetApiClient()

	Sub InternationalStreetExample()

		Dim lookup As New Lookup("Rua Padre Antonio D'Angelo 121 Casa Verde, Sao Paulo", "Brazil")
		With lookup
			.InputId = "ID-8675309"
			.Geocode = True
			.Organization = "John Doe"
			.Address1 = "Rua Padre Antonio D'Angelo 121"
			.Address2 = "Casa Verde"
			.Locality = "Sao Paulo"
			.AdministrativeArea = "SP"
			.Country = "Brazil"
			.PostalCode = "02516-050"
		End With

		Console.WriteLine("*******************************************************")
		Console.WriteLine()

		Try
			client.Send(lookup)
		Catch ex As SmartyException
			Console.WriteLine(ex.Message)
			Console.WriteLine(ex.StackTrace)
		Catch ex As IOException
			Console.WriteLine(ex.StackTrace)
		Catch ex As Exception
			Console.WriteLine(ex.Message)
			Console.WriteLine(ex.StackTrace)
		End Try

		Dim candidates = lookup.Result

		Console.WriteLine("Original lookup: " + lookup.Freeform + " " + lookup.Country)
		Console.WriteLine("Input ID: " + lookup.InputId + Environment.NewLine())

		If candidates.Count = 0 Then
			Console.WriteLine("No candidates. Tthe address is not valid." + Environment.NewLine)
			Return
		End If

		Console.WriteLine("Address is valid. (There is at least one candidate)")


		Console.WriteLine("*** Found " + CStr(candidates.Count) + " result" + If(candidates.Count = 1, "", "s") + " ***")

		For Each candidate In candidates
			Console.WriteLine()
			Dim components = candidate.Components
			Dim metadata = candidate.Metadata

			Console.WriteLine("Address: " + candidate.Analysis.VerificationStatus)
			Console.WriteLine("Address precision: " + candidate.Analysis.AddressPrecision + Environment.NewLine)

			Console.WriteLine("First Line: " + candidate.Address1)
			Console.WriteLine("Second Line: " + candidate.Address2)
			Console.WriteLine("Third Line: " + candidate.Address3)
			Console.WriteLine("Fourth Line: " + candidate.Address4)
			Console.WriteLine("Address Format: " + candidate.Metadata.AddressFormat)
			Console.WriteLine("Latitude: " + CStr(candidate.Metadata.Latitude))
			Console.WriteLine("Longitude: " + CStr(candidate.Metadata.Longitude))
		Next

		Console.WriteLine()



	End Sub

End Module
